### PR TITLE
Re-enable taiko edge effect and add fade

### PR DESCRIPTION
**Pull request by Joppe27**
Addresses https://github.com/ppy/osu/pull/20819#issuecomment-1288483388.

In this PR, the `EdgeEffect` immediately changes for hitobjects that are created after kiai time starts/ends (see `AccentColour` setter). The hitobjects that are already alive are handled by the `fadeEdgeEffects` function which fades them to their correct (non-)kiai value gracefully.

https://user-images.githubusercontent.com/56885706/200204413-782f035f-2787-4c87-b64b-916163c0a26e.mp4

### CODE REVIEW

File: osu.Game.Rulesets.Taiko/Skinning/Default/CirclePiece.cs [hunk 1]
```diff
@@ -50,25 +50,16 @@ public Color4 AccentColour
...
-        {
-            get => kiaiMode;
-            set
-            {
-                kiaiMode = value;
-
-                resetEdgeEffects();
-            }
-        }
+        private static bool kiaiMode;
```
Comment (peppy) [inline osu.Game.Rulesets.Taiko/Skinning/Default/CirclePiece.cs:62]: I'm 99.999% sure you didn't intend to make this static.

Comment (Joppe27) [reply osu.Game.Rulesets.Taiko/Skinning/Default/CirclePiece.cs]: Fixed!

**Comment by Joppe27** [comment]
I noticed an issue with this PR where the edge effect of hit objects that are already alive doesn't fade in fast enough to match new hit objects. This is barely noticeable on high BPM beatmaps but looks jank on low BPM. I don't have the time to look into this at the moment, feel free to make changes if the solution is obvious. If not I will *try* to fix this issue during the weekend.

https://user-images.githubusercontent.com/56885706/200358395-28f431bd-1065-458a-ba66-6f4af606d486.mp4

**Comment by Joppe27** [comment]
I finally got around to fixing the issue I pointed out, is visually good to go now!

**Comment by peppy** [comment]
I don't know about this implementation. From what I can tell from reading through, your making the edge effect fade in for each hit separately on the first beat, but I'd probably argue that it should be applied from the exact point where the kiai timing section becomes active. Or maybe just always applied.

Put another way, I think this may be overcomplicating this.

**Comment by peppy** [comment]
I'm pretty sure that's not what you want, as that variable changes state over time.

You probably want to get `HitObject.Kiai` instead.

**Comment by Joppe27** [comment]
I'm using that change of state to determine when kiai time starts, which I need to fade the effects of objects that are already alive. Since the `HitObject.Kiai` value never changes I can only fade the effects in on creation without requiring extra complexity, which wasn't the intention. How would you use `HitObject.Kiai` for already alive objects in a useful way? I feel like this is going over my head, sorry!

**Comment by peppy** [comment]
I was thinking that kiai objects can probably have the effect applied from the point they enter the screen, rather than at the point kiai time is entered. Not sure. Maybe we can just drop this glow altogether and call it a day, I'd rather not overthink it since the new skin doesn't do this same effect.

**Pull request closed without merging**